### PR TITLE
Skills

### PR DESCRIPTION
**Pull request by EduardoChuks**
Skills and splash screen added.

### CODE REVIEW

File: app/src/main/java/com/belatrixsf/connect/ui/skills/Add/addSkillFragment.java [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+                             Bundle savedInstanceState) {
+        setHasOptionsMenu(true);
+        return inflater.inflate(R.layout.fragment_keywords_list, container, false);
+    }
+
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        initViews();
+        keywordsPresenter.getKeywords(false);
```
Comment (PedroCarrillo) [inline app/src/main/java/com/belatrixsf/connect/ui/skills/Add/addSkillFragment.java]: @icerrate we won't save state?

File: app/src/main/java/com/belatrixsf/connect/networking/retrofit/api/ServerPaths.java [hunk 2]
```diff
@@ -71,6 +72,10 @@
     String CATEGORY_KEYWORD_LIST = CATEGORY + "/keyword/list";
     String SUBCATEGORIES_BY_CATEGORY_ID = CATEGORY + "/{" + CATEGORY_ID + "}/subcategory/list";
 
+    String EMPLOYEE_SKILLS = EMPLOYEE + "/{" + EMPLOYEE_ID + "}/"+SKILLS+"/list";
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/networking/retrofit/api/ServerPaths.java]: spaces

File: app/src/main/java/com/belatrixsf/connect/services/server/CategoryServerService.java [hunk 3]
```diff
@@ -72,4 +74,30 @@ public ServiceRequest getKeywords(BelatrixConnectCallback<List<Keyword>> callbac
...
+    public ServiceRequest getKeywordsByEmployee(int employeeId, BelatrixConnectCallback<PaginatedResponse<Keyword>> callback) {
+        Call<PaginatedResponse<Keyword>> call = categoryAPI.getKeywordsByEmployee(employeeId);
+        ServiceRequest<PaginatedResponse<Keyword>> serviceRequest = new ServerServiceRequest<>(call);
+        enqueue(serviceRequest, callback);
+        return serviceRequest;
+    }
+
+    @Override
+    public ServiceRequest saveKeywordToEmployee(int employeeId, String keywordName, BelatrixConnectCallback<Keyword> callback) {
+        SaveEmployeeKeywordRequest saveEmployeeKeywordRequest = new SaveEmployeeKeywordRequest(keywordName);
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/services/server/CategoryServerService.java]: Use common name Class

File: app/src/main/java/com/belatrixsf/connect/ui/SplashActivity.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+/**
+ * Created by echuquilin on 11/08/16.
+ */
+public class SplashActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/ui/SplashActivity.java]: remove spaceeeeee

File: app/src/main/java/com/belatrixsf/connect/ui/account/expanded/ExpandPictureFragment.java [hunk 5]
```diff
@@ -102,7 +102,7 @@ protected void initDependencies(BelatrixConnectApplication belatrixConnectApplic
     @Override
     public void showProfilePicture(final String profilePicture) {
         if (pictureImageView != null) {
-            ImageFactory.getLoader().loadFromUrl(
+            ImageFactory.getLoader().loadFromUrlWithoutCC(
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/ui/account/expanded/ExpandPictureFragment.java]: set scaleType as a parameter (Maybe set an enum on Loader class)

File: app/src/main/java/com/belatrixsf/connect/ui/skills/Add/AddSkillActivity.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        setContentView(R.layout.activity_base);
+        if (savedInstanceState == null) {
+            replaceFragment(AddSkillFragment.newInstance(), false);
+        }
+        setNavigationToolbar();
+    }
+
+    @Override
+    protected void navigateBack() {
+        SkillsListFragment.refreshFromAddSkill(); //fragment already exists
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/ui/skills/Add/AddSkillActivity.java]: Delete comment or put it over the line code

File: app/src/main/java/com/belatrixsf/connect/ui/skills/Add/addSkillFragment.java (hunk 1 above)
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/ui/skills/Add/addSkillFragment.java]: D: we MUST!

File: app/src/main/java/com/belatrixsf/connect/ui/skills/SkillsListFragment.java [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+import java.util.List;
+
+import butterknife.Bind;
+
+/**
+ * Created by echuquilin on 4/08/16.
+ */
+public class SkillsListFragment extends BelatrixConnectFragment implements SkillsListView, SkillListAdapter.KeywordListener {
+
+    private static SkillsListPresenter skillsListPresenter;
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/ui/skills/SkillsListFragment.java:44]: remove static property

File: app/src/main/java/com/belatrixsf/connect/ui/skills/SkillsListFragment.java [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        initViews();
+        skillsListPresenter.getSkills(PreferencesManager.get().getEmployeeId(),false);
+    }
+
+    private void initViews() {
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getActivity());
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/ui/skills/SkillsListFragment.java:116]: Create sub functions for adapter and button

File: app/src/main/java/com/belatrixsf/connect/utils/SnackbarUtils.java [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+import android.view.View;
+
+import com.belatrixsf.connect.R;
+
+/**
+ * Created by echuquilin on 11/08/16.
+ */
+public class SnackbarUtils {
+
+    public static Snackbar createInformationSnackBar(String message, View view, View.OnClickListener positiveListener) {
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/utils/SnackbarUtils.java]: Add param actionLabel

File: app/src/main/java/com/belatrixsf/connect/utils/media/loaders/GlideLoader.java [hunk 10]
```diff
@@ -86,17 +91,22 @@ public void loadFromPath(String path, ImageView imageView, ImageTransformation t
...
             Context context,
             DrawableTypeRequest<T> load,
             ImageTransformation transformation,
             final Callback callback,
-            Drawable placeholder
+            Drawable placeholder,
+            boolean hasCenterCrop
     ) {
-        load.centerCrop();
+        if(hasCenterCrop){
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/utils/media/loaders/GlideLoader.java]: Use an enum

File: app/src/main/java/com/belatrixsf/connect/utils/media/loaders/PicassoLoader.java [hunk 11]
```diff
@@ -39,7 +38,7 @@
 
     @Override
     public void loadFromBitmap(byte[] bitmapImg, ImageView imageView, ImageTransformation transformation, Callback callback, Drawable placeholder) {
-
+        //nothing
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/utils/media/loaders/PicassoLoader.java]: //TODO: Future implementation

File: app/src/main/res/layout/fragment_keywords_list.xml [hunk 12]
```diff
@@ -1,23 +1,47 @@
...
+                android:padding="@dimen/dimen_4_5"
+                android:text="@string/skills_button_add"
+                android:textColor="@color/colorAccent"
+                android:visibility="gone" />
+
+            <android.support.v7.widget.RecyclerView
+                android:id="@+id/keywords"
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                tools:context="com.belatrixsf.connect.ui.keywords.SearchingKeywordsFragment" />
```
Comment (icerrate) [inline app/src/main/res/layout/fragment_keywords_list.xml]: remove it

File: app/src/main/res/layout/fragment_skills_list.xml [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+        android:layout_height="match_parent"
+        android:background="@color/white">
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:orientation="vertical">
+
+            <Button
+                android:id="@+id/button_add_skill"
```
Comment (icerrate) [inline app/src/main/res/layout/fragment_skills_list.xml]: name of the action (without view type)

File: app/src/main/java/com/belatrixsf/connect/networking/retrofit/api/ServerPaths.java [hunk 14]
```diff
@@ -71,6 +72,10 @@
...
+    String EMPLOYEE_ADD_SKILL = EMPLOYEE + "/{" + EMPLOYEE_ID + "}/"+SKILLS+"/add/";
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/networking/retrofit/api/ServerPaths.java]: spaces

File: app/src/main/java/com/belatrixsf/connect/networking/retrofit/api/ServerPaths.java [hunk 15]
```diff
@@ -71,6 +72,10 @@
...
+    String EMPLOYEE_REMOVE_SKILL = EMPLOYEE + "/{" + EMPLOYEE_ID + "}/"+SKILLS+"/remove/";
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/networking/retrofit/api/ServerPaths.java]: spaces

File: app/src/main/java/com/belatrixsf/connect/services/server/CategoryServerService.java [hunk 16]
```diff
@@ -72,4 +74,30 @@ public ServiceRequest getKeywords(BelatrixConnectCallback<List<Keyword>> callbac
...
+        Call<Keyword> keywordCall = categoryAPI.saveEmployeeSkill(employeeId, saveEmployeeKeywordRequest);
+        ServiceRequest<Keyword> serviceRequest = new ServerServiceRequest<>(keywordCall);
+        enqueue(serviceRequest, callback);
+        return serviceRequest;
+    }
+
+    @Override
+    public ServiceRequest removeEmployeeKeyword(int employeeId, String keywordName, BelatrixConnectCallback<Keyword> callback) {
+        SaveEmployeeKeywordRequest saveEmployeeKeywordRequest = new SaveEmployeeKeywordRequest(keywordName);
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/services/server/CategoryServerService.java]: Use common name Class

File: app/src/main/java/com/belatrixsf/connect/utils/SnackbarUtils.java [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+        Snackbar snackbar = Snackbar
+                .make(view, message, Snackbar.LENGTH_LONG)
+                .setAction(R.string.dialog_option_confirm, positiveListener);
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/utils/SnackbarUtils.java]: Set dynamic actionLabel

File: app/src/main/java/com/belatrixsf/connect/utils/media/loaders/PicassoLoader.java [hunk 18]
```diff
@@ -85,6 +84,11 @@ public void loadFromPath(String path, ImageView imageView, ImageTransformation t
         load(context, Picasso.with(context).load(new File(path)), transformation, imageView, callback, placeholder);
     }
 
+    @Override
+    public void loadFromUrlWithoutCC(String url, ImageView imageView, ImageTransformation transformation, Callback callback, Drawable placeholder) {
+        //nothing
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/utils/media/loaders/PicassoLoader.java]: //TODO: Future implementation

File: app/src/main/java/com/belatrixsf/connect/adapters/ContactsKeywordListAdapter.java [hunk 19]
```diff
@@ -78,7 +78,8 @@ public void onBindDataViewHolder(RecyclerView.ViewHolder holder, int position) {
             ImageFactory.getLoader().loadFromUrl(employee.getAvatar(),
                     contactKeywordViewHolder.photoImageView,
                     ImageLoader.ImageTransformation.BORDERED_CIRCLE,
-                    contactKeywordViewHolder.photoImageView.getResources().getDrawable(R.drawable.contact_placeholder)
+                    contactKeywordViewHolder.photoImageView.getResources().getDrawable(R.drawable.contact_placeholder),
+                    ImageLoader.ScaleType.CENTERCROP
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/adapters/ContactsKeywordListAdapter.java]: CENTER_CROP

File: app/src/main/java/com/belatrixsf/connect/ui/account/edit/EditAccountFragment.java [hunk 20]
```diff
@@ -89,18 +89,11 @@
...
-    EditText skypeIdEditText;
-    @Bind(R.id.locationRadioGroup)
-    RadioGroup locationRadioGroup;
-    @Bind(R.id.edit_image)
-    ImageView editPictureImageView;
+    @Bind(R.id.profile_picture) ImageView pictureImageView;
+    @Bind(R.id.firstName) EditText firstNameEditText;
+    @Bind(R.id.lastName) EditText lastNameEditText;
+    @Bind(R.id.skypeId) EditText skypeIdEditText;
+    @Bind(R.id.locationRadioGroup) RadioGroup locationRadioGroup;@Bind(R.id.edit_image) ImageView editPictureImageView;
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/ui/account/edit/EditAccountFragment.java]: 2 lines

File: app/src/main/java/com/belatrixsf/connect/ui/account/edit/EditAccountFragment.java [hunk 21]
```diff
@@ -89,18 +89,11 @@
...
 
     private EditAccountPresenter editAccountPresenter;
     private String mProfilePicturePath;
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/ui/account/edit/EditAccountFragment.java]: remove "m"

File: app/src/main/java/com/belatrixsf/connect/ui/account/expanded/ExpandPictureFragment.java [hunk 22]
```diff
@@ -117,7 +117,8 @@ public void onFailure() {
                             startPostponedEnterTransition();
                         }
                     },
-                    null
+                    null,
+                    ImageLoader.ScaleType.FITCENTER
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/ui/account/expanded/ExpandPictureFragment.java]: FIT_CENTER

File: app/src/main/java/com/belatrixsf/connect/utils/SnackbarUtils.java [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+import android.support.design.widget.Snackbar;
+import android.view.View;
+
+/**
+ * Created by echuquilin on 11/08/16.
+ */
+public class SnackbarUtils {
+
+    public static Snackbar createInformationSnackBar(String message, String actionLabel, View view, View.OnClickListener positiveListener) {
+        Snackbar snackbar = Snackbar
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/utils/SnackbarUtils.java]: It looks like an unnecessary var assignment

File: app/src/main/java/com/belatrixsf/connect/utils/media/loaders/ImageLoader.java [hunk 24]
```diff
@@ -40,22 +40,27 @@
         CIRCLE
     }
 
-    void loadFromBitmap(byte[] bitmapImg, ImageView imageView, ImageTransformation transformation, Callback callback, Drawable placeholder);
+    enum ScaleType {
+        CENTERCROP,
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/utils/media/loaders/ImageLoader.java]: CENTER_CROP

File: app/src/main/java/com/belatrixsf/connect/utils/media/loaders/ImageLoader.java [hunk 25]
```diff
@@ -40,22 +40,27 @@
...
+        FITCENTER
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/utils/media/loaders/ImageLoader.java]: FIT_CENTER

File: app/src/main/java/com/belatrixsf/connect/utils/media/loaders/PicassoLoader.java [hunk 26]
```diff
@@ -38,49 +37,49 @@
 public class PicassoLoader implements ImageLoader {
 
     @Override
-    public void loadFromBitmap(byte[] bitmapImg, ImageView imageView, ImageTransformation transformation, Callback callback, Drawable placeholder) {
-
+    public void loadFromBitmap(byte[] bitmapImg, ImageView imageView, ImageTransformation transformation, Callback callback, Drawable placeholder, ImageLoader.ScaleType scaleType) {
+        //TODO future implementation
```
Comment (icerrate) [inline app/src/main/java/com/belatrixsf/connect/utils/media/loaders/PicassoLoader.java]: TODO: Future Implementation

File: app/src/main/res/layout/fragment_account.xml [hunk 27]
```diff
@@ -1,261 +1,268 @@
...
-                android:focusableInTouchMode="true"
-                android:orientation="vertical">
+                android:layout_height="match_parent">
 
                 <LinearLayout
                     android:layout_width="match_parent"
-                    android:layout_height="wrap_content"
+                    android:layout_height="match_parent"
                     android:background="@color/white"
+                    android:focusableInTouchMode="true"
```
Comment (icerrate) [inline app/src/main/res/layout/fragment_account.xml]: for?

File: app/src/main/res/layout/fragment_account.xml [hunk 28]
```diff
@@ -1,261 +1,268 @@
...
-                    android:layout_width="match_parent"
-                    android:layout_height="?android:attr/listPreferredItemHeightSmall"
-                    android:gravity="center"
-                    android:text="@string/no_data_found"
-                    android:visibility="gone"
-                    tools:visibility="visible"/>
+                </LinearLayout>
 
-            </LinearLayout>
+                <!-- Wrapping Imageview in a FrameLayout as a
```
Comment (icerrate) [inline app/src/main/res/layout/fragment_account.xml]: Remove comment

File: app/src/main/java/com/belatrixsf/connect/utils/media/loaders/ImageLoader.java (hunk 24 above)
Comment (EduardoChuks) [inline app/src/main/java/com/belatrixsf/connect/utils/media/loaders/ImageLoader.java]: CENTER_CROP

**Pull request merged**